cryptoconditions/types/preimage.py

```diff
@@ -0,0 +1,173 @@
+        cryptographically secure pseudo-random number generator (CSPRNG) can be
+        used to avoid having to store each individual preimage.
+
+        Args:
+             preimage: Secret data that will be hashed to form the condition.
+        """
+        if preimage and (not isinstance(preimage, bytes)):
+            raise TypeError('Preimage must be bytes, was: {}'.format(preimage))
+        self.preimage = preimage
+        self.asn1 = asn1
+
+    @property
+    def bitmask(self):
+        return self.FEATURE_BITMASK
+
+    #def write_hash_payload(self, hasher):
+    #    """
+    #    Generate the contents of the condition hash.
+
+    #    Writes the contents of the condition hash to a Hasher. Used internally by `getCondition`.
+
+    #    HASH = SHA256(PREIMAGE)
+
+    #    Args:
+    #         hasher (Hasher): Destination where the hash payload will be written.
+    #    """
+    #    if not isinstance(hasher, Hasher):
+    #        raise TypeError('hasher must be a Hasher instance')
+    #    if self.preimage is None:
+    #        raise ValueError('Could not calculate hash, no preimage provided')
+    #    hasher.write(self.preimage)
+
+    #def parse_payload(self, reader, payload_size):
+    #    """
+    #    Parse the payload of a SHA256 hashlock fulfillment.
+
+    #    Read a fulfillment payload from a Reader and populate this object with that fulfillment.
+
+    #    FULFILLMENT_PAYLOAD =
+    #        VARBYTES PREIMAGE
+
+    #    Args:
+    #        reader (Reader): Source to read the fulfillment payload from.
+    #        payload_size (int): Total size of the fulfillment payload.
+    #    """
+    #    if not isinstance(reader, Reader):
+    #        raise TypeError('reader must be a Reader instance')
+    #    self.preimage = reader.read(payload_size)
+
+    #def write_payload(self, writer):
+    #    """
+    #    Generate the fulfillment payload.
+
+    #    This writes the fulfillment payload to a Writer.
+
+    #    Args:
+    #        writer (Writer): Subject for writing the fulfillment payload.
+    #    """
+    #    if not isinstance(writer, (Writer, Predictor)):
+    #        raise TypeError('writer must be a Writer instance')
+    #    if self.preimage is None:
+    #        raise ValueError('Preimage must be specified')
+
+    #    writer.write(self.preimage)
+    #    return writer
+
+    def to_dict(self):
+        """
+        Generate a dict of the fulfillment
+
+        Returns:
+            dict: representing the fulfillment
+        """
+        return {
+            'type': 'fulfillment',
+            'type_id': self.TYPE_ID,
+            'bitmask': self.bitmask,
+            'preimage': self.preimage.decode()
+        }
+
+    @property
+    def asn1_dict_payload(self):
+        return {'preimage': self.preimage}
+
+    def to_asn1_dict(self):
+        return {PreimageSha256.TYPE_ASN1: self.asn1_dict_payload}
+
+    @property
+    def fingerprint_contents(self):
+        if self.preimage is None:
```

Comment:
`preimage` could be made into a `property` with the validation done in a `setter`. I would be more consistent with the implementation of the other subtypes and would remove the need for these checks